app.rb

```diff
@@ -3,46 +3,81 @@
 require 'sidekiq'
 Dotenv.load
 
-require_relative 'workers/worker'
+require_relative 'workers/cloudflare_challenge_worker'
 
-get '/certificate_generation/new/:domain' do
-  authenticate!
-  token = SecureRandom.hex
-  Worker.perform_async(params[:domain], params[:subdomains], params[:debug], params[:app_name], token)
-  content_type :json
+$redis = Redis.new(url: ENV['REDIS_URL'])
 
-  { status_path: "#{request.env['rack.url_scheme']}://#{request.env['HTTP_HOST']}/certificate_generation/#{token}" }.to_json
+before do
+  request.body.rewind
+  @request_payload = JSON.parse request.body.read
 end
 
-get '/certificate_generation/:token' do
-  authenticate!
+post '/certificate_request' do
   content_type :json
+  authenticate!
+  if params_valid?
+    status 200
+    token = SecureRandom.hex
+    CloudflareChallengeWorker.perform_async(@request_payload["zone"], @request_payload["domains"], token, true)
+    { status: 'queued', uuid: token }.to_json
+  else
+    status 422
+  end
+end
 
-  pipe = Sidekiq.redis do |conn|
-    conn.pipelined do
-      conn.get("#{params[:token]}_status")
-      conn.get("#{params[:token]}_error")
-      conn.get("#{params[:token]}_domain")
-      conn.get("#{params[:token]}_subdomains")
-      conn.get("#{params[:token]}_message")
-    end
+get '/certificate_request/:token' do
+  content_type :json
+  authenticate!
+  if $redis.exists("status_#{params["token"]}")
+    return status 200
   end
+  status 404
+  { status: 'token1234 not a valid token' }.to_json
+end
+
+private
 
-  {
-    token:      params[:token],
-    status:     pipe[0],
-    error:      pipe[1],
-    domain:     pipe[2],
-    subdomains: pipe[3].to_s.split(','),
-    message:    pipe[4]
-  }.to_json
+def params_valid?
+  !@request_payload["domains"].empty? && !@request_payload["heroku_app_name"].empty? && !@request_payload["zone"].empty?
```

Comment:
`.any?` might be more readable

```(@request_payload["domains"] + @request_payload["heroku_app_name"] + @request_payload["zone"]).any?``` might also work but might not be as obvious what it's doing